src/resources/templates/npl-instructions.md

```diff
@@ -0,0 +1,617 @@
+
+// With named arguments
+var namedInstance = ProtocolName[party1 = alice, party2 = bob](
+  initialValue = 42, textParameter = "example"
+);
+```
+
+## Permissions
+
+```npl
+@api  // Required for API instantiation
+protocol[issuer, recipient] Transfer() {
+  /**
+   * Allows the issuer to send money.
+   * @param amount Amount to send.
+   * @return Success status.
+   */
+  permission[issuer] sendMoney(amount: Number) returns Boolean {
+    return true;
+  };
+
+  /**
+   * Allows either party to check balance.
+   * @return Current balance.
+   */
+  permission[issuer | recipient] checkBalance() returns Number {
+    return 100;
+  };
+};
+```
+
+### Require Conditions
+
+```npl
+permission[sender] transfer(amount: Number) {
+  require(amount > 0, "Amount must be positive");
+  require(balance >= amount, "Insufficient balance");
+  // Permission body
+};
+```
+
+## States
+
+```npl
+@api  // Required for API instantiation
+protocol[buyer, seller] Purchase() {
+  initial state created;
+  state pending;
+  state shipped;
+  final state delivered;
+  final state cancelled;
+};
+```
+
+### States as Guards and Transitions
+
+```npl
+// Permission only valid in created or pending states
+permission[buyer] cancelOrder() | created, pending {
+  become cancelled;
+};
+
+// Transition to shipped state
+permission[seller] shipOrder() | pending {
+  become shipped;
+};
+```
+
+## Protocol State Methods
+
+1. **activeState() returns Optional**: Always unwrap with `getOrFail()` or `getOrElse()`:
+
+   ```npl
+   // CORRECT
+   if (protocol.activeState().getOrFail() == MyProtocol.States.pending) {
+     // Do something
+   };
+   ```
+
+2. **Handling state checks safely**:
+   ```npl
+   // Using getOrElse or checking presence
+   var state = protocol.activeState().getOrElse(MyProtocol.States.created);
+   ```
+
+## Standard Library
+
+NPL has a defined standard library. **Never invent or assume the existence of methods that aren't documented.**
+
+### Available Types
+
+1. **Basic Types**: `Boolean`, `Number`, `Text`, `DateTime`, `LocalDate`, `Duration`, `Period`, `Blob`, `Unit`
+2. **Collection Types**: `List<T>`, `Set<T>`, `Map<K, V>`
+3. **Complex Types**: `Optional<T>`, `Pair<A, B>`, `Party`, `Test`
+4. **User-Defined Types**: `Enum`, `Struct`, `Union`, `Identifier`, `Symbol`, `Protocol`
+
+### Standard Library Functions
+
+1. **Logging**: `debug()`, `info()`, `error()`
+2. **Constructors**: `listOf()`, `setOf()`, `mapOf()`, `optionalOf()`, `dateTimeOf()`, `localDateOf()`
+3. **Time and Duration**: `now()`, `millis()`, `seconds()`, `minutes()`, `hours()`, `days()`, `weeks()`, `months()`,
+   `years()`
+
+### Allowed Methods by Type
+
+Use ONLY these methods - do not hallucinate or invent others:
+
+1. **Collection Methods (all collections)**:
+
+   - `allMatch()`, `anyMatch()`, `contains()`, `flatMap()`, `fold()`, `forEach()`, `isEmpty()`, `isNotEmpty()`
+   - `map()`, `noneMatch()`, `size()`, `asList()`
+   - Collections of `Number`: `sum()`
```

Comment:
One could imagine smarter ways to keep the stdlib in sync than doing it manually, but that would be too much scope for now